src/clusterfuzz/_internal/base/tasks.py

```diff
@@ -412,6 +457,36 @@ def lease(self, _event=None):  # pylint: disable=arguments-differ
     track_task_end()
 
 
+class UTaskMainPubSubTask(PubSubTask):
+
+  def __init__(self, pubsub_message):
+    self._pubsub_message = pubsub_message
+    grandparent_class = super(PubSubTask, self)
+    grandparent_class.__init__(
+        self.attribute('command'),
+        self.attribute('argument'),
+        self.attribute('job'),
+        input_url=self.attribute('input_url'))
+
+
+def initialize_task(message):
+  """Creates a task from |messages|."""
+  if message.attributes.get('eventType', None) == 'OBJECT_FINALIZE':
+    # Handle postprocess task.
+    # The google cloud API for pub/sub notifications uses the data field unlike
+    # ClusterFuzz which uses attributes more.
+    data = json.loads(message.data)
+    name = data['name']
+    bucket = data['bucket']
+    output_url_argument = storage.get_cloud_storage_file_path(bucket, name)
+    return PostprocessPubSubTask(output_url_argument, message)
+
+  if message.attributes.get('input_url', None) is not None:
+    return UTaskMainPubSubTask(message)
+
+  return PubSubTask(message)
+
+
 class PostprocessPubSubTask(PubSubTask):
```

Comment:
Optional: it seems you might also be able to get rid of this class (and the grandparent class initialization dance) by instead initializating a `PubSubTask`, then overwriting the `command`, `job`  and `eta` fields. Or, by adding these as optional arguments to `PubSubTask.__init__()`.